test/json_test.cpp

```diff
@@ -0,0 +1,248 @@
+}
+
+TEST_CASE(empty_value_rev)
+{
+    std::string json_str = "null";
+    migraphx::value v    = migraphx::from_json_string(json_str);
+    migraphx::value ev   = migraphx::value();
+    EXPECT(v == ev);
+}
+
+TEST_CASE(empty_array)
+{
+    migraphx::value v;
+    migraphx::value arr = {v, v};
+    auto json_str       = migraphx::to_json_string(arr);
+    EXPECT(json_str == "[null,null]");
+}
+
+TEST_CASE(empty_array_rev)
+{
+    std::string json_str = "[null,null]";
+    migraphx::value v    = migraphx::from_json_string(json_str);
+    migraphx::value e;
+    migraphx::value ev = {e, e};
+    EXPECT(ev == v);
+}
+
+TEST_CASE(empty_object1)
+{
+    migraphx::value val = migraphx::from_json_string("{}");
+    EXPECT(val == migraphx::value::object{});
+    EXPECT(migraphx::to_json_string(migraphx::value::object{}) == "{}");
+}
+
+TEST_CASE(empty_array1)
+{
+    migraphx::value val = migraphx::from_json_string("[]");
+    EXPECT(val == migraphx::value::array{});
+    EXPECT(migraphx::to_json_string(migraphx::value::array{}) == "[]");
+}
+
+TEST_CASE(int_value)
+{
+    migraphx::value v    = -1;
+    std::string json_str = migraphx::to_json_string(v);
+    EXPECT(json_str == "-1");
+}
+
+TEST_CASE(int_value_rev)
+{
+    std::string json_str = "-1";
+    migraphx::value v    = migraphx::from_json_string(json_str);
+    migraphx::value ev   = -1;
+    EXPECT(v == ev);
+}
+
+TEST_CASE(unsigned_value)
+{
+    migraphx::value v    = 1;
+    std::string json_str = migraphx::to_json_string(v);
+    EXPECT(json_str == "1");
+}
+
+TEST_CASE(unsigned_value_rev)
+{
+    std::string json_str = "1";
+    migraphx::value v    = migraphx::from_json_string(json_str);
+    EXPECT(v.is_uint64());
+    EXPECT(v.get_uint64() == 1);
+}
+
+TEST_CASE(float_value)
+{
+    migraphx::value v    = 1.5;
+    std::string json_str = migraphx::to_json_string(v);
+    EXPECT(json_str == "1.5");
+}
+
+TEST_CASE(float_value_rev)
+{
+    std::string json_str = "1.5";
+    migraphx::value v    = migraphx::from_json_string(json_str);
+    migraphx::value ev   = 1.5;
+    EXPECT(v == ev);
+}
+
+TEST_CASE(array_value)
+{
+    migraphx::value v    = {1, 2};
+    std::string json_str = migraphx::to_json_string(v);
+    EXPECT(json_str == "[1,2]");
+}
+
+TEST_CASE(array_value_rev)
+{
+    std::string json_str = "[1,2]";
+    migraphx::value v    = migraphx::from_json_string(json_str);
+    EXPECT(v.is_array());
+    EXPECT(v.size() == 2);
+    EXPECT(v[0].get_uint64() == 1);
+    EXPECT(v[1].get_uint64() == 2);
+}
+
+TEST_CASE(object_value)
+{
+    migraphx::value v    = {{"a", 1.2}, {"b", true}};
+    std::string json_str = migraphx::to_json_string(v);
+    EXPECT(json_str == "{\"a\":1.2,\"b\":true}");
+}
+
+TEST_CASE(object_value_rev)
+{
+    std::string json_str = R"({"a":1.2,"b":true})";
+    migraphx::value v    = migraphx::from_json_string(json_str);
+    migraphx::value ev   = {{"a", 1.2}, {"b", true}};
+    EXPECT(v == ev);
+}
+
+TEST_CASE(empty_object)
```

Comment:
This test case should probably be called `null_object` and not empty, since it isn't empty.